src/ERa/ERaApi.hpp

```diff
@@ -821,6 +824,126 @@ void ERaApi<Proto, Flash>::processVirtualPinRequest(const ERaDataBuff& arrayTopi
     item = nullptr;
 }
 
+#if defined(ESP32)
+const char* scanWiFi() {
+    int nets = WiFi.scanNetworks(false, true, false, 150);
+    if (nets < 0) {
+        ERA_LOG("WiFi", "Failed to scan networks");
+        return nullptr;
+    }
+
+    static ERaJson json;
+	json.reset();
+    for (int i = 0; i < nets; ++i) {
+        json[i]["ssid"] = WiFi.SSID(i).c_str();
+        json[i]["rssi"] = WiFi.RSSI(i);
+        json[i]["bssid"] = WiFi.BSSIDstr(i).c_str();
+        json[i]["channel"] = WiFi.channel(i);
+        switch (WiFi.encryptionType(i)) {
+            case wifi_auth_mode_t::WIFI_AUTH_WEP:
+                json[i]["encryption"] = "WEP";
+                break;
+            case wifi_auth_mode_t::WIFI_AUTH_WPA_PSK:
+                json[i]["encryption"] = "WPA/PSK";
+                break;
+            case wifi_auth_mode_t::WIFI_AUTH_WPA2_PSK:
+                json[i]["encryption"] = "WPA2/PSK";
+                break;
+            case wifi_auth_mode_t::WIFI_AUTH_WPA_WPA2_PSK:
+                json[i]["encryption"] = "WPA/WPA2/PSK";
+                break;
+            case wifi_auth_mode_t::WIFI_AUTH_OPEN:
+                json[i]["encryption"] = "OPEN";
+                break;
+            default:
+                json[i]["encryption"] = "UNKNOWN";
+                break;
+        }
+    }
+
+    WiFi.scanDelete();
+    return json.getString();
+}
+
+bool tryConnectWifi(const char* newSsid, const char* newPassword) {
+    WiFi.disconnect();
+
+    if (newPassword) {
+        WiFi.begin(newSsid, newPassword);
+    } else {
+        WiFi.begin(newSsid);
+    }
+
+    for (int i = 0; i < 100; i++) { // 10s
+        if (WiFi.isConnected()) {
+            return true;
+        }
+        delay(100);
+    }
+    return false;
+}
+
+bool changeWifi(const char* newSsid, const char* newPassword) {
+    if (tryConnectWifi(newSsid, newPassword)) {
+        return true;
+    }
+    // tryConnectWifi(ERaConfig.ssid, ERaConfig.pass);  try connect old wifi
```

Comment:
pls add this logic, otherwise the chip will reset and no `fail` message

Reply:
Yeah, ESP32 v3.0.x has some bugs related to the watchdog, but it will automatically reconnect to the previous Wi-Fi if it fails.

Reply:
the thing is after connecting to old wifi, it need to send the {success: 0} message to mqtt

Reply:
Yeah, after reconnecting to the previous Wi-Fi, it will resend the message {success: 0} (tested).